addons/point_of_sale/models/pos_config.py

```diff
@@ -46,6 +46,9 @@ def _get_group_pos_manager(self):
     def _get_group_pos_user(self):
         return self.env.ref('point_of_sale.group_pos_user')
 
+    def _default_tip_product_id(self):
+        return self.env['product.product'].search([('default_code', '=', 'TIPS')])
```

Comment:
I don't know if this is very relevant but isn't better to use the `xmlId` instead of the `default_code` ? We can see with @caburj. 

The default code is `product_product_tip`

Reply:
Or maybe not, I don't really know what to do in that case, I'm interested to have the answer.

Reply:
@loukwets @davidmonnom  xml ID points to a record. But this record is not immune from deletion. So, if you try to get the value of that xml ID using `env.ref`, it's possible that you don't get a record. But indeed, MODA has a good point. Since we have "product_product_tip" as part of data set when installing pos, we can use that by default. But if not found, we look for other TIPS product as default.

Reply:
One more thing. Only take the first item in the search result otherwise it might crash because tip_product_id is many2one.